libs/core/algorithms/tests/unit/algorithms/min_element.cpp

```diff
@@ -126,6 +127,7 @@ void min_element_test()
 {
     test_min_element<std::random_access_iterator_tag>();
     test_min_element<std::forward_iterator_tag>();
+    test_min_element_semantics<std::random_access_iterator_tag>();
```

Comment:
Don't you have to forward declare this function? Also, why not calling it for forward-iterators as well?